src/main/java/seedu/us/among/model/endpoint/Address.java

```diff
@@ -19,8 +21,10 @@
     /*
      * The first character of the address must not be a whitespace,
      * otherwise " " (a blank string) becomes a valid input.
+     * Adapted from https://stackoverflow.com/a/163398/13725861
      */
-    public static final String VALIDATION_REGEX = "[^\\s].*";
+    public static final Pattern VALIDATION_REGEX =
+            Pattern.compile("^(https?|ftp|file)://[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]");
```

Comment:
I think the match for `ftp` and `file` are not necessary since they are not protocols we support.

Reply:
Alright will remove the two